src/libs/actions/ReimbursementAccount/setupWithdrawalAccount.js

```diff
@@ -306,8 +223,19 @@ function setupWithdrawalAccount(params) {
                 setFreePlanVerifiedBankAccountID(responseACHData.bankAccountID);
             }
 
-            if (currentStep === CONST.BANK_ACCOUNT.STEP.REQUESTOR) {
-                checkDataAndMaybeStayOnRequestorStep(responseACHData, getNextStep(updatedACHData));
+            const nextStep = getNextStep(updatedACHData);
+
+            // Some steps require several substeps based on some data.
+            // This logic is done php side: when we need to display a specific step with specific data, php adds them in the response in nextStepValues
+            // Example 1: When forcing manual step after adding Chase bank account via Plaid, so we can ask for the real numbers instead of the plaid substitutes
+            // Example 2: When on the requestor step, showing Onfido view after submitting the identity and retrieving the sdkToken
+            if (_.has(responseACHData, 'nextStepValues')) {
+                navigation.goToWithdrawalAccountSetupStep(_.get(responseACHData.nextStepValues, 'currentStep') || nextStep, {
+                    ...updatedACHData,
+                    ...responseACHData,
+                    ...responseACHData.nextStepValues,
```

Comment:
I haven't checked out the other PR yet, but is the intention to have the `nextStepValues` overwrite the `achData` (because `nextStepValues` are similar to `achData`)? And if so should we remove the `nextStepValues` after they are copied here? Or is that not necessary?

Reply:
Yes, that's the intention. Removing nextStepValues then is not necessary, but might be cleaner, so I can do it :)